rust/macros/lib.rs

```diff
@@ -144,3 +145,46 @@ pub fn module(ts: TokenStream) -> TokenStream {
 pub fn vtable(attr: TokenStream, ts: TokenStream) -> TokenStream {
     vtable::vtable(attr, ts)
 }
+
+/// Concatenate two identifiers.
+///
+/// This is useful in macros that need to declare or reference items with names
+/// starting with a fixed prefix and ending in a user specified name. The resulting
+/// identifier has the span of the second argument.
```

Comment:
Wouldn't be better to have the span of the proc macro invocation?

Reply:
That would be confusing if the caller of `concat_idents!` is invoked from another macro, but the identifier originates from outside of this macro.

Reply:
Oh, I see. In the example below it would point to one of the `BR_` identifiers.

This also assumes that we are prefixing identifiers, but I think it's ok as that's the only usage ATM.